src/Core/Tokens/DataProtectorTokenFactory.cs

```diff
@@ -45,8 +48,9 @@ public bool TryUnprotect(string token, out T data)
             data = Unprotect(token);
             return true;
         }
-        catch
+        catch (Exception ex)
         {
+            _logger.LogInformation(ex, "Failed to unprotect token: {0}", token);
```

Comment:
No harm in doubling up I guess, but won't this already be caught and logged in `Token.UnprotectWith`?

Reply:
No, there are several steps to token validation and at least two (remove prefix and unprotect) can through.